src/game/inventory/inventory_func.c

```diff
@@ -14,6 +14,11 @@ bool Inv_AddItem(const GAME_OBJECT_ID object_id)
 {
     if (Object_IsObjectType(object_id, g_GunObjects)) {
         Gun_UpdateLaraMeshes(object_id);
+        if (g_Lara.gun_type == LGT_UNARMED) {
+            g_Lara.gun_type = Gun_GetType(object_id);
+            g_Lara.gun_status = LGS_ARMLESS;
+            Gun_InitialiseNewWeapon();
```

Comment:
Is this call necessary?

Reply:
Yes otherwise Lara's arms are screwed up when drawing her weapon especially the shotgun. Setting `LGS_ARMLESS` before the call prevents the relevant `DrawMeshes` from being called that would add the weapon to her hand mesh. This call is used in the `Lara_InitialiseInventory`, `Gun_Control` when drawing a weapon, and `Lara_RevertToPistolsIfNeeded`. I don't think it's an issue?

Reply:
Maybe I can call `Gun_Pistols_Ready` or `Gun_Rifle_Ready` instead? Those are only called during `Gun_X_Draw` functions though, so I didn't want the code to look confusing.

Reply:
No it's fine as-is, thanks for explaining.